### ISSUE TITLE
FacesConverter not instantiated at startup

### ISSUE THREAD

**Issue by Warkdev** [OpenLiberty/open-liberty#9725]
**Describe the bug**
FacesConverter cannot be injected within my JSF xhtml page although it seems to be found at startup but not instantiated. It keeps displaying the following stacktrace:

```
java.lang.NullPointerException
	at org.apache.myfaces.cdi.converter.FacesConverterCDIWrapper.getAsString(FacesConverterCDIWrapper.java:62)
	at org.apache.myfaces.shared.renderkit.RendererUtils.getStringValue...
```
I do have a beans.xml within WEB-INF containing:
`<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://xmlns.jcp.org/xml/ns/javaee"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_1_1.xsd" 
	bean-discovery-mode="all">
</beans>`

I do have AppConfig bean containing:
`@FacesConfig(version = FacesConfig.Version.JSF_2_3)`

And I do have a managed FacesConverter annotation:
`@FacesConverter(value = "accountTypeConverter",				
				managed = true)`

**Steps to Reproduce**
- Create a default Java EE 8 application.
- Create a dummy page displaying objects which have converter applied to:
`<f:facet name="output"><h:outputText value="#{account.fkAccountType}" converter="accountTypeConverter"/></f:facet>`
- Create named FacesConverter matching the converter defined above.
- Try to run the sample, a NPE should be raised by CDI. At best, I've managed to get a "null" injection within my backing bean. It seems that the FacesConverter isn't created at application startup.

**Expected behavior**
Converter should be working as designed and be instantiated and injected within the backing bean.This seems to be unrelated to Apache MyFaces 2.3 as I could get this working on Apache TomEE PluME 8.0.0 before.

**Diagnostic information:**
 - OpenLiberty Version: 19.0.0.11
 - Java Version: openjdk version "13.0.1" 2019-10-15
 - server.xml configuration:

```
<server description="DEV Budget Manager">

    <!-- Enable features -->
    <featureManager>
		<feature>javaee-8.0</feature>
	    <feature>localConnector-1.0</feature>
    </featureManager>

    <!-- This template enables security. To get the full use of all the capabilities, a keystore and user registry are required. -->
    
    <!-- For the keystore, default keys are generated and stored in a keystore. To provide the keystore password, generate an 
         encoded password using bin/securityUtility encode and add it below in the password attribute of the keyStore element. 
         Then uncomment the keyStore element. -->    
    <keyStore id="defaultKeyStore" password="<dummy>"/>     
    
    <!--For a user registry configuration, configure your user registry. For example, configure a basic user registry using the
        basicRegistry element. Specify your own user name below in the name attribute of the user element. For the password, 
        generate an encoded password using bin/securityUtility encode and add it in the password attribute of the user element. 
        Then uncomment the user element. -->
    
    <!-- To access this server from a remote client add a host attribute to the following element, e.g. host="*" -->
    <httpEndpoint httpPort="<port>" httpsPort="<https port>" id="defaultHttpEndpoint">
    </httpEndpoint>

    <!-- Automatically expand WAR files and EAR files -->
    <applicationManager autoExpand="true"/>


    <applicationMonitor updateTrigger="mbean"/>
    <jpa defaultJtaDataSourceJndiName="jdbc/<my_jdbc>" entityManagerPoolCapacity="5"/>
    <library id="MARIADBLIB" name="Library for Maria DB">
        
    <fileset dir="${shared.resource.dir}" includes="mariadb-java-client-2.5.1.jar"/>
    </library>
    <dataSource id="budgetDB" jndiName="jdbc/BudgetDBDS">
    	<jdbcDriver javax.sql.DataSource="org.mariadb.jdbc.Driver" libraryRef="MARIADBLIB"/>
    	<properties URL="jdbc:mariadb://<local_ip>/<local_db>" databaseName="<local>" password="<dummy>" portNumber="<local port>" serverName="<local>" user="<user>"/>
        
        
        
    </dataSource>    
    <webAppSecurity singleSignonEnabled="false"/>
    <basicRegistry id="basic" realm="dummyRealm">
    	<user name="admin" password="<who cares>"/>
    </basicRegistry>

    <webApplication id="BudgetManager" location="BudgetManager.war" name="BudgetManager"/>
</server>
```

**Additional context**
- Using JSF 2.3
- Using Primefaces 7.0
[messages.log](https://github.com/OpenLiberty/open-liberty/files/3832215/messages.log)

**Comment by Warkdev** [comment]
The same is happening with a simple TEST page and the same converter.

Side note, if the converter is tagged with both value and forClass, it can be injected into other backing beans. If not, the following error message is appearing:

```
[ERROR   ] CWWKZ0004E: Une exception s'est produite lors du démarrage de l'application BudgetManager. Le message d'exception était : com.ibm.ws.container.service.state.StateChangeException: org.jboss.weld.exceptions.DeploymentException: WELD-001409: Ambiguous dependencies for type AccountTypeConverter with qualifiers @FacesConverter
  at injection point [BackedAnnotatedField] @Inject @FacesConverter private eu.bm.views.users.AccountBean.converter
  at eu.bm.views.users.AccountBean.converter(AccountBean.java:0)
  Possible dependencies: 
  - Managed Bean [class eu.bm.views.users.converters.AccountTypeConverter] with qualifiers [@Any @FacesConfig @FacesConverter],
  - org.apache.myfaces.cdi.converter.FacesConverterProducer@28b1ca9
```

**Comment by Warkdev** [comment]
And after a few debugging, it seems that the bean is available within the CDI container:

`Set<Bean<?>> beans = beanManager.getBeans(Object.class,new AnnotationLiteral<Any>() {}));
        for (Bean<?> bean : beans) {
            System.out.println(bean.getBeanClass().getName());
        }`

Produces:

` 20:53:01.955 [Default Executor-thread-424] DEBUG eu.bm.views.users.TestConverterBean - bean: eu.bm.views.users.converters.AccountTypeConverter`

So, what's wrong in this lookup ?

**Comment by Warkdev** [comment]
You can retrieve the following WAR file and test it against Open Liberty to verify the behaviour: https://ufile.io/hs518rtd

![image](https://user-images.githubusercontent.com/7893254/68700352-0b80c880-0585-11ea-9255-a28a9baf6562.png)

**Comment by Warkdev** [comment]
Is apparently linked to myfaces see here:https://stackoverflow.com/questions/58825333/facesconverter-is-null-jsf-2-3-cdi-2-0/58886671?noredirect=1#

**Comment by volosied** [comment]
Originally, I said this should be fixed when JSF 2.3.6, but I believe it will now be when 2.3.7 is pulled into liberty.
